File: backend/AiConsultant/src/main/java/com/zx/consultant/document/entity/Document.java
package com.zx.consultant.document.entity;

import lombok.Data;
import java.time.LocalDateTime;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.FieldFill;

/**
 * 文档实体类
 * Document
 */
@Data
@TableName("document")
public class Document {
    
    /**
     * 主键 ID
     */
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /** 关联的知识库ID */
    private Long knowledgeId;

    private String fileName;

    /** 文件类型：PDF, MD, TXT */
    private String fileType;

    /** 文件大小（字节） */
    private Long fileSize;

    /** 磁盘/OSS存储路径 */
    private String storagePath;

    /** 状态：UPLOADING, PARSING, INDEXING, READY, FAILED */
    private String status;

    /** 切块数量 */
    private Integer chunkCount;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend/src/types/document.ts
import type { DocumentStatus } from '@/constants/enum'

export interface DocumentItem {
  id: number | string
  knowledgeId: number | string
  /** 后端字段：文件名 */
  fileName: string
  fileType?: string
  fileSize?: number
  storagePath?: string
  status: DocumentStatus
  chunkCount?: number
  createTime?: string
  updateTime?: string
}

export interface UploadParams {
  knowledgeId: number | string
  file: File
}

/** POST /api/v1/documents/upload 响应 data */
export interface UploadResult {
  documentId: number | string
  status: DocumentStatus
}

/** GET /api/v1/documents 查询参数 */
export interface DocumentListQuery {
  knowledgeId: number | string
  page?: number
  /** 每页条数（后端参数名为 size） */
  size?: number
}

/** GET /api/v1/documents/{documentId}/status 响应 data */
export interface DocumentStatusResult {
  status: DocumentStatus
}
